docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt

```diff
@@ -51,9 +49,122 @@ Resource budget:
  Details
 *********
 
+Conceptually, the ``adaptive_asha`` searcher is a carefully tuned
```

Comment:
@liamcli this is a mash-up of all the previous adaptive guides, but rewritten a bit to remove the stuff that was wrong from sha to asha. i removed the old adaptive guides. if you could, either review this section or just take a pass after this lands to add corrections (that just may be faster than you reviewing it old, or honestly push some fixes to my branch if you want)